### ISSUE TITLE
All end points should be plural?

### ISSUE THREAD

**Issue by c5sire** [plantbreeding/API#275]
There are still a few endpoints that are not in the plural, like:

- layout -> layouts
- vendor -> vendors

There are a few words that are usually deemed to be plural in itself but do have also correct plural forms like:
- progeny -> progenies (https://www.merriam-webster.com/dictionary/progeny)
- pedigree -> pedigrees (http://www.learnersdictionary.com/definition/pedigree)
- germplasm -> germplasms (https://www.wordhippo.com/what-is/the-plural-of/germplasm.html)

**Comment by BrapiCoordinatorSelby** [comment]
TL;DR I will change layout to layouts, the others I have reasons for keeping as they are
sorry for the essay, I'm highly opinionated about these things

---

This rule of "everything must be plural" usually makes sense because in most cases you are returning a collection of objects. But there are always exceptions to this type of rule. An API is all about communicating between developers, so we should strive to make that communication clear, straightforward and intuitive. I would rather it be clear what each call does rather than follow an arbitrary rule. To put it another way, someone new to BrAPI shouldn't need to know the plurals rule to make sense of any of the calls. So I would recommend we make everything plural by default, unless there is a clear reason to make an exception for clarity. 

Layout vs layouts is a good catch, I just missed that one. The only thing that gives me pause is the data returned in the layout call is a collection of plant/plot positions. So is that collection considered 1 layout or is that a collection of layouts or should it be /plantpositions instead to be plural. Since its not 100% clear either way, I'm happy to follow the rule change this to "vendors"

vendor I left singular on purpose. Vendor is not an object in the BrAPI schema. It more represents a category and is used to group a collection of related calls. I was worried about the need to differentiate generic samples and vendor samples (vendors samples is wrong). So vendor acts more as an adjective than the plural noun in question. 
This explanation makes sense to me, but I will happily pluralize vendors if the community feels strongly about it.

For progeny and pedigree, I would argue that these terms are for more intuitive than their technically plural counter parts. The pedigree call returns only 1 pedigree tree, and the progeny call returns only 1 set of progeny. "Progenies" and "pedigrees" implies multiple pedigree trees or multiple sets of progeny could be returned. Again, I will happily change these if the community feels strongly about it.

For germplasm, I will reference the first line of the BrAPI Germplasm documentation:
> Fun Fact: The plural of germplasm is germplasm (no "s").
I don't know who originally wrote it there, but apparently they thought about this question of plural germplasm and left a note about it hahaha
